mozci/task.py

```diff
@@ -164,10 +165,35 @@ def update_group(result):
                 if not is_bad_group(self.id, result.group)
             ]
 
+        # I'm storing in the cache in here rather than at the end of _load_errorsummary()
+        # because there's various modifications of various self properties.
```

Comment:
This means we will store this data even when we already have it in the cache from AD.
What if we stored it in _load_errorsummary, after the call to __post__init__? Or, you could store the raw values and then call __post__init__ also when the data is coming from the cache

Reply:
I've adjusted the code to put it after the **post_Iinit** call. Thanks!

What should we do when there's data from AD?